### DISCUSSION TITLE
What's the simplest way to change the font?

### DISCUSSION THREAD

**Discussion by pedrommcarrasco** [gonzalezreal/swift-markdown-ui#293] [Q&A]
Hey 👋,

I'm in search of the most straightforward method to modify the font and font size globally within the Markdown element. Despite reviewing documentation and conducting various experiments, I've hit a roadblock in identifying a simple solution for this adjustment.

Any guidance or suggestions would be greatly appreciated!

**Comment by gonzalezreal** [comment]
Hi @pedrommcarrasco,

To configure the base font family and size on a `Markdown` view, you can use the [`markdownTextStyle(_:textStyle:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/markdown/markdowntextstyle(_:textstyle:)) view modifier and override the current theme [`text`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/theme/text) text style.

```swift
Markdown {
  """
  ## Try MarkdownUI

  **MarkdownUI** is a native Markdown renderer for SwiftUI
  compatible with the
  [GitHub Flavored Markdown Spec](https://github.github.com/gfm/).
  """
}
.markdownTextStyle(\.text) {
  FontFamily(.custom("Avenir"))
  FontSize(20)
}
```

You can combine different text styles like [`FontFamily`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/fontfamily), [`FontFamilyVariant`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/fontfamilyvariant), [`FontStyle`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/fontstyle), [`FontSize`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/fontsize), [`FontWeight`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.3.0/documentation/markdownui/fontweight), etc.

You should always use an absolute size when overriding the default text style. Other text styles like headings or code blocks normally use relative sizes, based on the default text size.

**Reply by pedrommcarrasco** [reply]
Thank you @gonzalezreal 👍
What about defining how the font should scale with dynamic type? For example, which TextStyle (body, subheadline, etc...) I want it to be based on?

**Reply by gonzalezreal** [reply]
Dynamic type in MarkdownUI works by scaling text relative to the `body` text style. There is currently no API to change that.

If you want to pick a font size that works well in all platforms, you can use `FontProperties.defaultSize`, which is the default if you don't specify a font size. I hope that makes sense 😄 .

**Reply by yousefNowPay** [reply]
Hi @gonzalezreal  thanks for the amazing library I have question can we change the font size in case of markdownTextStyle(\.emphasis) ? something like this ` .markdownTextStyle(\.emphasis) {
                BackgroundColor(Color.yellow500.opacity(0.3))
                FontSize(fontSize)
            }`
cause these didn't work
